lib/appsignal/helpers/instrumentation.rb

```diff
@@ -5,6 +5,144 @@ module Helpers
+      #
+      # This helper is recommended to be used in Ruby scripts and parts of an
+      # app not already instrumented by AppSignal's automatic instrumentations.
+      #
+      # Use this helper in combination with our {.instrument} helper to track
+      # instrumentation events.
+      #
+      # If AppSignal is not active ({Appsignal.active?}) it will still execute
+      # the block, but not create a transaction for it.
+      #
+      # @example Instrument a block of code
+      #   Appsignal.monitor(
+      #     :namespace => "my_namespace",
+      #     :action => "MyClass#my_method"
+      #   ) do
+      #     # Some code
+      #   end
+      #
+      # @example Instrument a block of code with an instrumentation event
+      #   Appsignal.monitor(
+      #     :namespace => "my_namespace",
+      #     :action => "MyClass#my_method"
+      #   ) do
+      #     Appsignal.instrument("some_event.some_group") do
+      #       # Some code
+      #     end
+      #   end
+      #
+      # @example Set custom metadata on the transaction
+      #   Appsignal.monitor(
+      #     :namespace => "my_namespace",
+      #     :action => "MyClass#my_method"
+      #   ) do
+      #     # Some code
+      #
+      #     Appsignal.set_tags(:tag1 => "value1", :tag2 => "value2")
+      #     Appsignal.set_params(:param1 => "value1", :param2 => "value2")
+      #   end
+      #
+      # @example Call monitor within monitor (not recommended)
+      #   Appsignal.monitor(
+      #     :namespace => "my_namespace",
+      #     :action => "MyClass#my_method"
+      #   ) do
+      #     # This will _not_ update the namespace and action name
+      #     Appsignal.monitor(
+      #       :namespace => "my_other_namespace",
+      #       :action => "MyOtherClass#my_other_method"
+      #     ) do
+      #       # Some code
+      #
+      #       # The reported namespace will be "my_namespace"
+      #       # The reported action will be "MyClass#my_method"
+      #     end
+      #   end
+      #
+      # @param namespace [String/Symbol] The namespace to set on the new
+      #   transaction.
+      #   Defaults to the 'web' namespace.
+      #   This will update the already active transaction's namespace if
+      #   {.monitor} is called when another transaction is already active.
+      # @param action [String/Symbol] The action name for the transaction.
+      #   This will update the already active transaction's action if
+      #   {.monitor} is called when another transaction is already active.
+      # @yield The block to monitor.
+      # @raise [Exception] Any exception that occurs within the given block is
+      #   re-raised by this method.
+      # @return [Object] The value of the given block is returned.
+      # @since 3.11.0
+      def monitor(
+        namespace: nil,
+        action: nil
+      )
+        return yield unless active?
+
+        has_parent_transaction = Appsignal::Transaction.current?
```

Comment:
It seems to me that calling `monitor` when there's already a transaction in progress has no effect -- it only sets the error it returns in the transaction, which is something the parent transaction should already be doing anyway. Maybe we can just `return yield if has_parent_transaction` early on?